File: src/test/java/io/binac/leetcode/LongestSubstringWithoutRepeatingCharactersTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class LongestSubstringWithoutRepeatingCharactersTest {
    private final LongestSubstringWithoutRepeatingCharacters.Solution1 solution1 = new LongestSubstringWithoutRepeatingCharacters.Solution1();

    @Test
    void test1() {
        assertEquals(3, solution1.lengthOfLongestSubstring("abcabcbb"));
        assertEquals(1, solution1.lengthOfLongestSubstring("bbbbb"));
        assertEquals(3, solution1.lengthOfLongestSubstring("pwwkew"));
    }

    @Test
    void test1End() {
        assertEquals(2, solution1.lengthOfLongestSubstring("au"));
    }

    @Test
    void test1Duplicate() {
        assertEquals(2, solution1.lengthOfLongestSubstring("abba"));
    }
}

File: cpp/tests/LongestSubstringWithoutRepeatingCharactersTest.cpp
#include "LongestSubstringWithoutRepeatingCharacters.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(LongestSubstringWithoutRepeatingCharacters, Solution3_1) {
  auto S3_1 = Solution3_1();

  EXPECT_EQ(3, S3_1.lengthOfLongestSubstring("abcabcbb"));
  EXPECT_EQ(1, S3_1.lengthOfLongestSubstring("bbbbb"));
  EXPECT_EQ(3, S3_1.lengthOfLongestSubstring("pwwkew"));
  EXPECT_EQ(0, S3_1.lengthOfLongestSubstring(""));
}